src/components/App/PuzzleGenerate/index.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import styles from './PuzzleGenerate.module.scss';
+import { Button } from '../../Global';
+import { getAllPuzzles } from '../../../services';
+import {createPuzzleInstance} from "../../../services/puzzleInstance";
+
+const PuzzleGenerate = () => {
+  const handleSubmit = event => {
+    event.preventDefault();
+    const hint = event.target[0].value;
+    const latitude = event.target[1].value;
+    const longitude = event.target[2].value;
+    const address = event.target[3].value;
+    const puzzle = event.target[4].value;
+
+    console.log(hint, latitude, longitude, address, puzzle);
+    createPuzzleInstance(puzzle, longitude, latitude, address, hint).then(r => console.log(r));
+  };
+
+  const [puzzleList, setPuzzleList] = useState([]);
+
+  useEffect(() => {
+    const fetchPuzzles = async () => {
```

Comment:
I've seen this pattern somewhere before but why do we do it like this again? Would it be bad to define `fetchPuzzles()` outside of the hook's body and then just call the function within the hook's body?

Reply:
I'm not sure about this but I think if you put this function inside of useEffect, it will not rerender when the component rerender.

Reply:
I'll look into this later